ambry-router/src/test/java/com.github.ambry.router/DefaultGCMCryptoServiceTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.utils.TestUtils;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+import java.util.Properties;
+import javax.crypto.spec.SecretKeySpec;
+import org.bouncycastle.util.encoders.Hex;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static com.github.ambry.router.CryptoTestUtils.*;
+
+
+/**
+ * Tests {@link DefaultGCMCryptoService} and {@link DefaultCryptoServiceFactory}
+ */
+public class DefaultGCMCryptoServiceTest {
```

Comment:
Do you have a test for deserializing different IV sizes with the same crypto service?

Reply:
good point. Will add it.